lib/game.rb

```diff
@@ -0,0 +1,50 @@
+require_relative './player.rb'
+require_relative './board.rb'
+require_relative './ui.rb'
+
+class Game
+  include UI
+  def initialize
+    board = Board.new
+    @players = []
+    output('Welcome to Tic-tac-toe!', true)
+    2.times { |n| get_player_data(n) }
+    @current_player = @players[0]
+    @other_player = @players[1]
+    while !won? || !full?
+      clear_console
+      output(players_info(),true)
+      board.draw()
+      ask_move(@current_player)
+      switch_players()
+    end
+  end
+
+  def get_player_data(n)
+    output("Enter Player #{n + 1} name: ")
+    name = input
+    name = name.empty? ? "Player #{n + 1}" : name
+    sign = n.even? ? 'X' : 'O'
+    @players[n] = Player.new(name, sign)
+  end
+
+  def players_info
```

Comment:
1.) Game appears to hold too many responsibilities. This method appears to be more of in the Player class's domain
2. This method mixes UI logic with your objects. Let's avoid this